registry/rate-limits.go

```diff
@@ -53,9 +53,9 @@ func (d *rateLimitedRetryingRoundTripper) RoundTrip(req *http.Request) (*http.Re
 			doRetry = true // TODO maximum number of retries? (perhaps a deadline instead?  req.WithContext to inject a deadline?  👀)
 		}
 
-		// 503 should result in a few auto-retries (especially with the automatic retry delay this injects), but up to a limit so we don't contribute to the "thundering herd" too much in a serious outage
-		if res.StatusCode == 503 && maxTry503 > 1 {
-			maxTry503--
+		// certain status codes should result in a few auto-retries (especially with the automatic retry delay this injects), but up to a limit so we don't contribute to the "thundering herd" too much in a serious outage
+		if (res.StatusCode == 503 || res.StatusCode == 502 || res.StatusCode == 500) && maxTry50X > 1 {
```

Comment:
Why not just `(res.StatusCode >= 500 && res.StatusCode < 600)`?

Reply:
I thought about it, but hadn't gone through all the possible 5xx to make sure it was sane for us to retry on _all_ of them.  It seems unlikely for Hub to use one of the others unless there's a real error, but I'm sure just because I've put that in writing we'll start seeing it soon. :see_no_evil:

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status#server_error_responses

Reply:
If it starts responding with 418, we're in trouble